matrix-sdk/src/main/java/org/matrix/androidsdk/util/EventDisplay.java

```diff
@@ -162,18 +177,50 @@ public CharSequence getTextualDisplay(Integer displayNameColor) {
+                text = jsonEventContent.has("body") ? jsonEventContent.get("body").getAsString() : null;
 
                 // check for html formatting
                 if (jsonEventContent.has("formatted_body") && jsonEventContent.has("format")) {
                     String format = jsonEventContent.getAsJsonPrimitive("format").getAsString();
                     if (Message.FORMAT_MATRIX_HTML.equals(format)) {
                         String htmlBody = jsonEventContent.getAsJsonPrimitive("formatted_body").getAsString();
 
+                        if (mHtmlToolbox != null) {
+                            htmlBody = mHtmlToolbox.convert(htmlBody);
+                        }
+
+                        // Special treatment for "In reply to" message
+                        if (jsonEventContent.has("m.relates_to")) {
+                            JsonElement relatesTo = jsonEventContent.get("m.relates_to");
+                            if (relatesTo.isJsonObject()) {
+                                if (relatesTo.getAsJsonObject().has("m.in_reply_to")) {
+                                    // Replace <blockquote data-mx-reply><a href=\"__permalink__\">In reply to</a>
+                                    // By <blockquote data-mx-reply><a href=\"#\">['In reply to' from resources]</a>
+                                    // To disable the link and to localize the "In reply to" string
+                                    if (htmlBody.startsWith(MESSAGE_IN_REPLY_TO_FIRST_PART)) {
+                                        int index = htmlBody.indexOf(MESSAGE_IN_REPLY_TO_LAST_PART);
+
+                                        if (index != -1) {
+                                            htmlBody = MESSAGE_IN_REPLY_TO_FIRST_PART
+                                                    + "<a href=\"#\">"
+                                                    + mContext.getString(R.string.message_reply_to_prefix)
+                                                    + htmlBody.substring(index);
+                                        }
+                                    }
+                                }
+                            }
+                        }
+
                         // some markers are not supported so fallback on an ascii display until to find the right way to manage them
                         // an issue has been created https://github.com/vector-im/vector-android/issues/38
-                        if (!TextUtils.isEmpty(htmlBody) && !htmlBody.contains("<ol>") && !htmlBody.contains("<li>")) {
-                            text = Html.fromHtml(jsonEventContent.getAsJsonPrimitive("formatted_body").getAsString());
+                        // BMA re-enable <ol> and <li> support (https://github.com/vector-im/riot-android/issues/2184)
+                        if (!TextUtils.isEmpty(htmlBody) /*&& !htmlBody.contains("<ol>") && !htmlBody.contains("<li>")*/) {
```

Comment:
Note: You should remove the comments from the condition. Your comment above is enough

Reply:
ok, done